sessionctx/variable/session.go

```diff
@@ -3274,3 +3280,141 @@ func (lst *protectedTSList) Size() (size int) {
 	lst.Unlock()
 	return
 }
+
+// stmtSummary is an adapter to redirect executions to instances of different types of statement summary.
+type stmtSummary struct {
+	// EnablePersistent indicates whether to enable the persistence
+	// of statements summary in the current session.
+	EnablePersistent bool
+	// StmtSummaryV2 refers to the global StmtSummary instance that will be used when
+	// EnablePersistent is true.
+	StmtSummaryV2 *stmtsummaryv2.StmtSummary
+}
+
+func newStmtSummary() *stmtSummary {
+	return &stmtSummary{
+		EnablePersistent: config.GetGlobalConfig().Instance.StmtSummaryEnablePersistent,
+		StmtSummaryV2:    stmtsummaryv2.GlobalStmtSummary,
+	}
+}
+
+func (s *stmtSummary) Add(stmtExecInfo *stmtsummary.StmtExecInfo) {
+	if !s.ensureInit() {
+		return
```

Comment:
1. When will this happen？
2. If this happened, TiDB will lost all statement summary info?

Reply:
Its purpose is to provide "possibly necessary" initialization for v2, which only happens when persistence is enabled and GlobalStmtSummary is uninitialized, typically in the early stages of some internal session.